### ISSUE TITLE
`x + 0 == 5` does not simplify

### ISSUE THREAD

**Issue by joshhansen** [josdejong/mathjs#2461]
Ran into an unexpected simplify failure in some code I'm working. It's present on current develop, as well as 10.2.0:

```
> math.simplify('x + 7 - 7 == 12 - 7').toString()
'x + 0 == 5'
```

The parse tree is:

```
         ==
       /     \
     -        \
    /  \       \
  +     \       -
 /  \    \     /  \  
x   7    7     12  7
```

And it turns out that `x + 0 == 5` itself does not simplify:
```
> math.simplify('x + 0 == 5').toString()
'x + 0 == 5'
```

**Comment by gwhitney** [comment]
Nice catch Josh; the backstop simple recursion for simplifyCore on arbitrary binary nodes was stuck one level too deep in the code, so some cases were slipping through.

**Comment by josdejong** [comment]
Fix published in `v10.4.0`
